### PR TITLE
chore(trading): update dropdowns

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Issue: #4606 

# Description ℹ️

Updating trading dropdowns. 

# Demo 📺
Screenshot 2023-09-04 at 19 31 52
Screenshot 2023-09-04 at 19 31 33

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: apps/trading/components/market-selector/asset-dropdown.tsx [hunk 1]
```diff
@@ -20,22 +21,41 @@ export const AssetDropdown = ({
   checkedAssets: string[];
   onSelect: (id: string, checked: boolean) => void;
 }) => {
+  const assetsText = useMemo(() => {
```
Comment (MadalinaRaicu) [inline apps/trading/components/market-selector/asset-dropdown.tsx:24]: without this change the trading dropdown does not work

File: libs/markets/src/lib/markets-provider.ts [hunk 2]
```diff
@@ -199,7 +199,7 @@ export const allMarketsWithLiveDataProvider = makeDerivedDataProvider<
     return data.find(
       (market) =>
         market.id ===
-        (parts[1].delta as MarketDataUpdateFieldsFragment).marketId
+        (parts[1].delta as MarketDataUpdateFieldsFragment)?.marketId
```
Comment (MadalinaRaicu) [inline libs/markets/src/lib/markets-provider.ts:202]: not great but sorted out this issue 
Screenshot 2023-09-04 at 18 39 57

**Comment by mattrussell36** [comment]
@MadalinaRaicu The TradingDropdownTrigger component will no longer render a styled button. You'll need to a button element/component within the trigger.

**Comment by MadalinaRaicu** [comment]
> @MadalinaRaicu The TradingDropdownTrigger component will no longer render a styled button. You'll need to a button element/component within the trigger.

created a bespoke button component for the market selector.

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**